src/TypeScript/EslintBridgeClient/EslintBridgeHttpWrapper.cs

```diff
@@ -0,0 +1,96 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.Helpers;
+
+namespace SonarLint.VisualStudio.TypeScript.EslintBridgeClient
+{
+    internal interface IEslintBridgeHttpWrapper : IDisposable
+    {
+        Task<string> PostAsync(string serverEndpoint, object request = null);
+    }
+
+    [Export(typeof(IEslintBridgeHttpWrapper))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class EslintBridgeHttpWrapper : IEslintBridgeHttpWrapper
+    {
+        private readonly IEslintBridgeStartUp eslintBridgeStartUp;
+        private readonly ILogger logger;
+        private readonly HttpClient httpClient;
+
+        [ImportingConstructor]
+        public EslintBridgeHttpWrapper(IEslintBridgeStartUp eslintBridgeStartUp, ILogger logger)
+            : this(eslintBridgeStartUp, new HttpClientHandler(), logger)
+        {
+        }
+
+        internal EslintBridgeHttpWrapper(IEslintBridgeStartUp eslintBridgeStartUp, HttpMessageHandler httpHandler, ILogger logger)
+        {
+            this.eslintBridgeStartUp = eslintBridgeStartUp;
+            this.logger = logger;
+            httpClient = new HttpClient(httpHandler);
+        }
+
+        public async Task<string> PostAsync(string serverEndpoint, object request = null)
+        {
+            try
+            {
+                var serializedRequest = request == null ? string.Empty : JsonConvert.SerializeObject(request, Formatting.Indented);
+                logger.LogDebug(Resources.INFO_RequestDetails, serverEndpoint, serializedRequest);
+
+                var port = await eslintBridgeStartUp.Start();
+                logger.LogDebug(Resources.INFO_ServerStarted, port);
+
+                var content = new StringContent(serializedRequest, Encoding.UTF8, "application/json");
+                var response = await httpClient.PostAsync($"http://localhost:{port}/{serverEndpoint}", content);
+                var responseString = await response.Content.ReadAsStringAsync();
+
+                logger.LogDebug(Resources.INFO_ResponseDetails, serverEndpoint, responseString);
+
+                return responseString;
+            }
+            catch (AggregateException ex) 
+            {
+                var exceptions = string.Join(Environment.NewLine, ex.InnerExceptions);
```

Comment:
Is it worth dumping the stack trace if an exception occurs? The exception will be happening in framework code so I'm not sure it will tell us much.
I think it would better just to log the error type and message, and if we want the stack trace to log it as a debug message.
Same for the non-aggregate exception handler below.

Reply:
Update: see my comment in the other PR about whether it's worth catching exceptions here.

Reply:
I ended up leaving the try-catch in both places. The http wrapper catches aggregate exceptions and prints them nicely, and it's not something that I'd want to repeat in every API call, but the client still needs to handle its own exceptions, e.g. deserialization failures.